KeeAgent/KeeAgentInteractiveUi.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Threading;
+using System.Windows.Forms;
+
+namespace KeeAgent
+{
+  /// <summary>
+  /// Provides a separate UI thread for interactive communication without interlocking with the MainWindow.
+  /// </summary>
+  public sealed class KeeAgentInteractiveUi : IDisposable
+  {
+    private const string ComponentName = "KeeAgent Interactive UI";
+    private SynchronizationContext _synchronizationContext;
+    private ApplicationContext _applicationContext;
+
+    public KeeAgentInteractiveUi()
+    {
+      var uiThread = new Thread(UiThreadMain) {
+        Name = ComponentName,
+        IsBackground = true // active background thread will not block the application shutdown process
+      };
+      uiThread.SetApartmentState(ApartmentState.STA);
+      uiThread.Start();
+    }
+
+    private void UiThreadMain()
```

Comment:
I would rather have the required thread type here with the attribute.

```suggestion
    [STAThread]
    private void UiThreadMain()
```

Reply:
@dlech , this attribute is only for the `Main` entrypoint method. Please, refere to the [STAThreadAttribute Class](https://learn.microsoft.com/en-us/dotnet/api/system.stathreadattribute?view=net-8.0) documentation.

> Apply this attribute to the entry point method (the Main() method in C# and Visual Basic). **It has no effect on other methods.** To set the apartment state of threads you start in your code, use the [Thread.SetApartmentState](https://learn.microsoft.com/en-us/dotnet/api/system.threading.thread.setapartmentstate?view=net-8.0) or [Thread.TrySetApartmentState](https://learn.microsoft.com/en-us/dotnet/api/system.threading.thread.trysetapartmentstate?view=net-8.0) method before starting the thread.